### ISSUE TITLE
multiprocess cannot recognize nested functions?

### ISSUE THREAD

**Issue by alsorain** [uqfoundation/multiprocess#56]
After installing multiprocess for Python 3.7, I try to run the following code,

```
import numpy as np
from multiprocess import Pool

def f(x,y,z):
    return x+y+z

def ff(args):
    return f(*args)

il = [(1,2,3),(4,5,67),(9,9,9)]

if __name__ == '__main__':
    p = Pool(4)
    result = p.map(ff, il)
    print(result)
```
Then I get the error,
```

  File "C:/Dropbox/Projects/housing market/code/python Housing Market/multiprocessing_practice.py", line 17, in <module>
    result = p.map(ff, il)

  File "C:\tools\Anaconda3\lib\site-packages\multiprocess-0.70.8.dev0-py3.7.egg\multiprocess\pool.py", line 657, in get
    raise self._value

NameError: name 'f' is not defined
```
A similar error pops up when I running the example file. 
It is strange. Or, did I make some mistakes?

**Comment by yesthisistom** [comment]
I've got this also.  The best current answer I have is this:

```
def funct_1(x):
    def funct_2(x):
        return x**2

    return funct_2(x)
p = Pool(4)
result = p.map(funct1, [1,2,3,4)
```
You also have to have imports included in that function.  It's not great.

**Comment by jetsnguns** [comment]
Experiencing the same issue with version 0.70.11.1. Is this the intended behavior?

**Comment by matho10** [comment]
@jetsnguns 
Hi just let you know that the problem is solved when I switch to ubuntu 20.04. Pathos works smoothly in linux environment (as well as MacOS) for me. There seems to be compatibility issues for pathos in windows. If you are using windows, this could help you.

**Comment by mmckerns** [comment]
@matho10: I also don't see an error when running the code (on a Mac).
The issue on Windows is that Windows does not provide a fork, so the maps on Windows require a more extensive pickling and some additional help to freeze the code. Long story short, you should use `freeze_support` on Windows to help aid the serialization across the map.

See this example (using nested functions) for how to use `freeze_support`:
https://github.com/uqfoundation/pathos/blob/master/examples/nested.py

Please reopen this issue if this doesn't solve it for you.
